flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -112,6 +114,13 @@ public TaskManagerRunner(
 		// Initialize the TM metrics
 		TaskExecutorMetricsInitializer.instantiateStatusMetrics(taskManagerMetricGroup, taskManagerServices.getNetworkEnvironment());
 
+		HeartbeatManagerImpl heartbeatManager = new HeartbeatManagerImpl(
+				taskManagerConfiguration.getTimeout().toMilliseconds(),
+				resourceID,
+				executor,
+				Executors.newSingleThreadScheduledExecutor(),
```

Comment:
Do you have further suggestions where to introduce the **ScheduledExecutorService** seems suitable?